Description: Ubuntu 24
 Patch Boost, Python and GCC to build on Ubuntu 24.
 .
 uhd (3.13.1.0-4) UNRELEASED; urgency=medium
 .
  * Patch for Boost 1.83
  * Patch for Python 3.12
  * Patch for GCC 13.3.0
  * Patch for Python 3
Author: Alexis Giraudet <alexis.giraudet@nav-timing.safrangroup.com>

--- uhd-3.13.1.0.orig/host/CMakeLists.txt
+++ uhd-3.13.1.0/host/CMakeLists.txt
@@ -238,6 +238,7 @@ if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GN
     add_definitions(-Wall)
     add_definitions(-Wextra)
     add_definitions(-Wsign-compare)
+    add_definitions(-Wno-error=format-security)
     #add_definitions(-Wconversion)
     #add_definitions(-pedantic)
     #add_definitions(-ansi)
@@ -368,8 +369,18 @@ if(${ENABLE_PYTHON3})
     set(BOOST_PYTHON_FOUND ${Boost_PYTHON37_FOUND})
     set(BOOST_PYTHON_LIBRARY ${Boost_PYTHON37_LIBRARY})
   endif()
+  if(${Boost_PYTHON310_FOUND})
+    set(BOOST_PYTHON_FOUND ${Boost_PYTHON310_FOUND})
+    set(BOOST_PYTHON_LIBRARY ${Boost_PYTHON310_LIBRARY})
+  endif()
+  if(${Boost_PYTHON312_FOUND})
+    set(BOOST_PYTHON_FOUND ${Boost_PYTHON312_FOUND})
+    set(BOOST_PYTHON_LIBRARY ${Boost_PYTHON312_LIBRARY})
+  endif()
 endif()
 
+add_compile_definitions(BOOST_TIMER_ENABLE_DEPRECATED)
+
 message(STATUS "Boost include directories: ${Boost_INCLUDE_DIRS}")
 message(STATUS "Boost library directories: ${Boost_LIBRARY_DIRS}")
 message(STATUS "Boost libraries: ${Boost_LIBRARIES}")
--- uhd-3.13.1.0.orig/host/include/uhd/utils/byteswap.ipp
+++ uhd-3.13.1.0/host/include/uhd/utils/byteswap.ipp
@@ -89,7 +89,7 @@
 /***********************************************************************
  * Define the templated network to/from host conversions
  **********************************************************************/
-#include <boost/detail/endian.hpp>
+#include <boost/endian.hpp>
 
 namespace uhd {
 
--- uhd-3.13.1.0.orig/host/lib/include/uhdlib/experts/expert_nodes.hpp
+++ uhd-3.13.1.0/host/lib/include/uhdlib/experts/expert_nodes.hpp
@@ -16,6 +16,7 @@
 #include <boost/thread/recursive_mutex.hpp>
 #include <boost/thread.hpp>
 #include <boost/units/detail/utility.hpp>
+#include <boost/core/noncopyable.hpp>
 #include <memory>
 #include <list>
 #include <stdint.h>
--- uhd-3.13.1.0.orig/host/lib/rfnoc/dma_fifo_block_ctrl_impl.cpp
+++ uhd-3.13.1.0/host/lib/rfnoc/dma_fifo_block_ctrl_impl.cpp
@@ -14,6 +14,7 @@
 #include <boost/make_shared.hpp>
 #include <boost/thread/mutex.hpp>
 #include <boost/format.hpp>
+#include <boost/bind.hpp>
 
 using namespace uhd;
 using namespace uhd::rfnoc;
--- uhd-3.13.1.0.orig/host/lib/transport/gen_vrt_if_packet.py
+++ uhd-3.13.1.0/host/lib/transport/gen_vrt_if_packet.py
@@ -23,7 +23,7 @@ TMPL_TEXT = """<% import time %>
 #include <uhd/exception.hpp>
 #include <uhd/transport/vrt_if_packet.hpp>
 #include <uhd/utils/byteswap.hpp>
-#include <boost/detail/endian.hpp>
+#include <boost/endian.hpp>
 #include <vector>
 
 //define the endian macros to convert integers
--- uhd-3.13.1.0.orig/host/lib/usrp/b100/b100_impl.cpp
+++ uhd-3.13.1.0/host/lib/usrp/b100/b100_impl.cpp
@@ -16,6 +16,7 @@
 #include <uhd/utils/safe_call.hpp>
 #include <uhdlib/usrp/common/apply_corrections.hpp>
 #include <boost/format.hpp>
+#include <boost/bind.hpp>
 #include <cstdio>
 #include <iostream>
 #include <chrono>
--- uhd-3.13.1.0.orig/host/lib/usrp/common/lmx2592.cpp
+++ uhd-3.13.1.0/host/lib/usrp/common/lmx2592.cpp
@@ -12,6 +12,7 @@
 #include <cmath>
 #include <iomanip>
 #include <iostream>
+#include <array>
 
 using namespace uhd;
 
--- uhd-3.13.1.0.orig/host/lib/usrp/cores/rx_dsp_core_3000.cpp
+++ uhd-3.13.1.0/host/lib/usrp/cores/rx_dsp_core_3000.cpp
@@ -15,6 +15,7 @@
 #include <boost/assign/list_of.hpp>
 #include <boost/thread/thread.hpp> //thread sleep
 #include <boost/math/special_functions/round.hpp>
+#include <boost/bind.hpp>
 #include <algorithm>
 #include <cmath>
 
--- uhd-3.13.1.0.orig/host/lib/usrp/cores/tx_dsp_core_3000.cpp
+++ uhd-3.13.1.0/host/lib/usrp/cores/tx_dsp_core_3000.cpp
@@ -14,6 +14,7 @@
 #include <boost/assign/list_of.hpp>
 #include <boost/math/special_functions/round.hpp>
 #include <boost/thread/thread.hpp> //sleep
+#include <boost/bind.hpp>
 #include <algorithm>
 #include <cmath>
 
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/db_cbx.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/db_cbx.cpp
@@ -8,6 +8,7 @@
 #include "db_sbx_common.hpp"
 #include <boost/algorithm/string.hpp>
 #include <boost/math/special_functions/round.hpp>
+#include <boost/bind.hpp>
 
 using namespace uhd;
 using namespace uhd::usrp;
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/db_dbsrx.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/db_dbsrx.cpp
@@ -23,6 +23,7 @@
 #include <boost/assign/list_of.hpp>
 #include <boost/format.hpp>
 #include <boost/math/special_functions/round.hpp>
+#include <boost/bind.hpp>
 #include <utility>
 #include <cmath>
 #include <chrono>
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/db_dbsrx2.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/db_dbsrx2.cpp
@@ -21,6 +21,7 @@
 #include <boost/format.hpp>
 #include <boost/thread.hpp>
 #include <boost/math/special_functions/round.hpp>
+#include <boost/bind.hpp>
 #include <utility>
 
 using namespace uhd;
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/db_sbx_common.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/db_sbx_common.cpp
@@ -6,6 +6,7 @@
 //
 
 #include "db_sbx_common.hpp"
+#include <boost/bind.hpp>
 
 using namespace uhd;
 using namespace uhd::usrp;
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/db_sbx_version3.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/db_sbx_version3.cpp
@@ -9,6 +9,7 @@
 #include "db_sbx_common.hpp"
 #include <uhd/types/tune_request.hpp>
 #include <boost/algorithm/string.hpp>
+#include <boost/bind.hpp>
 
 using namespace uhd;
 using namespace uhd::usrp;
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/db_sbx_version4.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/db_sbx_version4.cpp
@@ -9,6 +9,7 @@
 #include "db_sbx_common.hpp"
 #include <uhd/types/tune_request.hpp>
 #include <boost/algorithm/string.hpp>
+#include <boost/bind.hpp>
 
 using namespace uhd;
 using namespace uhd::usrp;
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/db_tvrx.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/db_tvrx.cpp
@@ -32,6 +32,7 @@
 #include <boost/thread.hpp>
 #include <boost/array.hpp>
 #include <boost/math/special_functions/round.hpp>
+#include <boost/bind.hpp>
 #include <utility>
 #include <cmath>
 #include <cfloat>
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/db_tvrx2.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/db_tvrx2.cpp
@@ -55,6 +55,7 @@
 #include <boost/format.hpp>
 #include <boost/array.hpp>
 #include <boost/math/special_functions/round.hpp>
+#include <boost/bind.hpp>
 #include <utility>
 #include <cmath>
 #include <chrono>
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/db_ubx.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/db_ubx.cpp
@@ -24,6 +24,7 @@
 #include <boost/math/special_functions/round.hpp>
 #include <boost/algorithm/string.hpp>
 #include <boost/thread/mutex.hpp>
+#include <boost/bind.hpp>
 #include <map>
 #include <chrono>
 #include <thread>
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/db_xcvr2450.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/db_xcvr2450.cpp
@@ -51,6 +51,7 @@
 #include <boost/assign/list_of.hpp>
 #include <boost/format.hpp>
 #include <boost/math/special_functions/round.hpp>
+#include <boost/bind.hpp>
 #include <utility>
 #include <chrono>
 #include <thread>
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/magnesium/magnesium_radio_ctrl_cpld.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/magnesium/magnesium_radio_ctrl_cpld.cpp
@@ -8,6 +8,7 @@
 #include "magnesium_cpld_ctrl.hpp"
 #include "magnesium_constants.hpp"
 #include <uhd/utils/log.hpp>
+#include <thread>
 
 using namespace uhd;
 using namespace uhd::usrp;
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/neon/neon_radio_ctrl_impl.cpp
+++ uhd-3.13.1.0/host/lib/usrp/dboard/neon/neon_radio_ctrl_impl.cpp
@@ -21,6 +21,7 @@
 #include <sstream>
 #include <cmath>
 #include <cstdlib>
+#include <thread>
 
 using namespace uhd;
 using namespace uhd::usrp;
--- uhd-3.13.1.0.orig/host/lib/usrp/multi_usrp.cpp
+++ uhd-3.13.1.0/host/lib/usrp/multi_usrp.cpp
@@ -28,6 +28,7 @@
 #include <bitset>
 #include <chrono>
 #include <thread>
+#include <boost/bind.hpp>
 
 using namespace uhd;
 using namespace uhd::usrp;
--- uhd-3.13.1.0.orig/host/lib/usrp/usrp2/usrp2_impl.cpp
+++ uhd-3.13.1.0/host/lib/usrp/usrp2/usrp2_impl.cpp
@@ -21,7 +21,8 @@
 #include <boost/bind.hpp>
 #include <boost/asio/ip/address_v4.hpp>
 #include <boost/asio.hpp> //used for htonl and ntohl
-#include <boost/thread.hpp>
+#include <boost/math/special_functions/round.hpp>
+#include <boost/math/special_functions/sign.hpp>
 
 using namespace uhd;
 using namespace uhd::usrp;
@@ -878,9 +879,6 @@ meta_range_t usrp2_impl::get_tx_dsp_freq
     );
 }
 
-#include <boost/math/special_functions/round.hpp>
-#include <boost/math/special_functions/sign.hpp>
-
 void usrp2_impl::update_clock_source(const std::string &mb, const std::string &source){
     //NOTICE: U2_REG_MISC_CTRL_CLOCK is on the wb clock, and cannot be set from fifo_ctrl
     //clock source ref 10mhz
--- uhd-3.13.1.0.orig/host/lib/usrp/x300/x300_radio_ctrl_impl.cpp
+++ uhd-3.13.1.0/host/lib/usrp/x300/x300_radio_ctrl_impl.cpp
@@ -19,6 +19,7 @@
 #include <boost/algorithm/string.hpp>
 #include <boost/make_shared.hpp>
 #include <boost/date_time/posix_time/posix_time_io.hpp>
+#include <boost/bind.hpp>
 #include <chrono>
 #include <thread>
 
--- uhd-3.13.1.0.orig/host/lib/usrp_clock/octoclock/octoclock_impl.cpp
+++ uhd-3.13.1.0/host/lib/usrp_clock/octoclock/octoclock_impl.cpp
@@ -13,6 +13,7 @@
 #include <boost/filesystem.hpp>
 #include <boost/format.hpp>
 #include <boost/thread.hpp>
+#include <boost/bind.hpp>
 
 #include <uhd/device.hpp>
 #include <uhd/exception.hpp>
--- uhd-3.13.1.0.orig/firmware/fx2/utils/build_eeprom.py
+++ uhd-3.13.1.0/firmware/fx2/utils/build_eeprom.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2004,2006 Free Software Foundation, Inc.
 # 
--- uhd-3.13.1.0.orig/firmware/fx2/utils/edit-gpif-b100.py
+++ uhd-3.13.1.0/firmware/fx2/utils/edit-gpif-b100.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- Python -*-
 #
 # Copyright 2003 Free Software Foundation, Inc.
--- uhd-3.13.1.0.orig/firmware/fx2/utils/edit-gpif.py
+++ uhd-3.13.1.0/firmware/fx2/utils/edit-gpif.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- Python -*-
 #
 # Copyright 2003 Free Software Foundation, Inc.
--- uhd-3.13.1.0.orig/firmware/fx2/utils/generate_regs.py
+++ uhd-3.13.1.0/firmware/fx2/utils/generate_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os, os.path
 import re
--- uhd-3.13.1.0.orig/firmware/usrp2/bin/bin_to_mif.py
+++ uhd-3.13.1.0/firmware/usrp2/bin/bin_to_mif.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import struct
 import sys
--- uhd-3.13.1.0.orig/firmware/usrp2/bin/bin_to_ram_macro_init.py
+++ uhd-3.13.1.0/firmware/usrp2/bin/bin_to_ram_macro_init.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import struct
 import sys
--- uhd-3.13.1.0.orig/firmware/usrp2/bin/divisors.py
+++ uhd-3.13.1.0/firmware/usrp2/bin/divisors.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 speeds = (9600, 19200, 38400, 57600, 115200, 230400)
 
--- uhd-3.13.1.0.orig/firmware/usrp2/bin/elf_to_sbf
+++ uhd-3.13.1.0/firmware/usrp2/bin/elf_to_sbf
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2009 Free Software Foundation, Inc.
 #
--- uhd-3.13.1.0.orig/firmware/usrp2/bin/serial_loader
+++ uhd-3.13.1.0/firmware/usrp2/bin/serial_loader
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2009 Free Software Foundation, Inc.
 #
--- uhd-3.13.1.0.orig/firmware/usrp2/bin/uart_ihex_flash_loader.py
+++ uhd-3.13.1.0/firmware/usrp2/bin/uart_ihex_flash_loader.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import serial
 from optparse import OptionParser
--- uhd-3.13.1.0.orig/firmware/usrp2/bin/uart_ihex_ram_loader.py
+++ uhd-3.13.1.0/firmware/usrp2/bin/uart_ihex_ram_loader.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import serial
 from optparse import OptionParser
--- uhd-3.13.1.0.orig/firmware/usrp3/n230/n230_burner.py
+++ uhd-3.13.1.0/firmware/usrp3/n230/n230_burner.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2014 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/firmware/usrp3/n230/n230_debug.py
+++ uhd-3.13.1.0/firmware/usrp3/n230/n230_debug.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010-2011 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/firmware/usrp3/utils/bin_to_coe.py
+++ uhd-3.13.1.0/firmware/usrp3/utils/bin_to_coe.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import sys
 import binascii
--- uhd-3.13.1.0.orig/firmware/usrp3/x300/x300_aurora_bist.py
+++ uhd-3.13.1.0/firmware/usrp3/x300/x300_aurora_bist.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2016 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/firmware/usrp3/x300/x300_debug.py
+++ uhd-3.13.1.0/firmware/usrp3/x300/x300_debug.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010-2014 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/fpga-src/usrp1/gen_makefile_extra.py
+++ uhd-3.13.1.0/fpga-src/usrp1/gen_makefile_extra.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2006 Free Software Foundation, Inc.
 # 
--- uhd-3.13.1.0.orig/fpga-src/usrp1/sdr_lib/gen_cordic_consts.py
+++ uhd-3.13.1.0/fpga-src/usrp1/sdr_lib/gen_cordic_consts.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import math
 
--- uhd-3.13.1.0.orig/fpga-src/usrp1/toplevel/mrfm/mrfm.py
+++ uhd-3.13.1.0/fpga-src/usrp1/toplevel/mrfm/mrfm.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # This is mrfm_fft_sos.py
 # Modification of Matt's mrfm_fft.py that reads filter coefs from file
--- uhd-3.13.1.0.orig/fpga-src/usrp1/toplevel/mrfm/mrfm_fft.py
+++ uhd-3.13.1.0/fpga-src/usrp1/toplevel/mrfm/mrfm_fft.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # This is mrfm_fft_sos.py
 # Modification of Matt's mrfm_fft.py that reads filter coefs from file
--- uhd-3.13.1.0.orig/fpga-src/usrp2/sdr_lib/gen_cordic_consts.py
+++ uhd-3.13.1.0/fpga-src/usrp2/sdr_lib/gen_cordic_consts.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import math
 
--- uhd-3.13.1.0.orig/fpga-src/usrp2/top/extract_usage.py
+++ uhd-3.13.1.0/fpga-src/usrp2/top/extract_usage.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2012 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/fpga-src/usrp2/top/python/check_inout.py
+++ uhd-3.13.1.0/fpga-src/usrp2/top/python/check_inout.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/fpga-src/usrp2/top/python/check_timing.py
+++ uhd-3.13.1.0/fpga-src/usrp2/top/python/check_timing.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2011 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/fpga-src/usrp3/build.py
+++ uhd-3.13.1.0/fpga-src/usrp3/build.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2016 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/fpga-src/usrp3/lib/crossbar/synth/synth_axis_ctrl_crossbar_2d_mesh.py
+++ uhd-3.13.1.0/fpga-src/usrp3/lib/crossbar/synth/synth_axis_ctrl_crossbar_2d_mesh.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python3
+#!/usr/bin/python3
 #!/usr/bin/python3
 #
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/fpga-src/usrp3/lib/crossbar/synth/synth_chdr_crossbar_nxn.py
+++ uhd-3.13.1.0/fpga-src/usrp3/lib/crossbar/synth/synth_chdr_crossbar_nxn.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python3
+#!/usr/bin/python3
 #!/usr/bin/python3
 #
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/fpga-src/usrp3/lib/crossbar/synth/synth_run.py
+++ uhd-3.13.1.0/fpga-src/usrp3/lib/crossbar/synth/synth_run.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python3
+#!/usr/bin/python3
 #!/usr/bin/python3
 #
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/fpga-src/usrp3/lib/io_port2/create-lvbitx.py
+++ uhd-3.13.1.0/fpga-src/usrp3/lib/io_port2/create-lvbitx.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 """
 Generate lvbitx from template and bitfile
 """
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/scripts/launch_vivado.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/scripts/launch_vivado.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Notice: Some parts of this file were copied from PyBOMBS, which has a
 # different copyright, and is highlighted appropriately. The following
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/scripts/uhd_image_builder.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/scripts/uhd_image_builder.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 Copyright 2016-2017 Ettus Research
 
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/scripts/uhd_image_builder_gui.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/scripts/uhd_image_builder_gui.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 Copyright 2016-2017 Ettus Research
 
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/scripts/viv_gen_ip_makefile.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/scripts/viv_gen_ip_makefile.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#!/usr/bin/python3
 
 import sys, os
 import collections
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/scripts/viv_gen_part_id.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/scripts/viv_gen_part_id.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import argparse
 import os, sys
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/scripts/viv_ip_xci_editor.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/scripts/viv_ip_xci_editor.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import argparse
 import os, sys
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/scripts/xil_bitfile_parser.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/scripts/xil_bitfile_parser.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import argparse
 import os, sys
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/utils/gen_xdc_from_rinf.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/utils/gen_xdc_from_rinf.py
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#!/usr/bin/python3
 
 import sys, os
 import collections
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/utils/rfnoc-system-sim/colosseum_models.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/utils/rfnoc-system-sim/colosseum_models.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2016 Ettus Research
 #
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/utils/rfnoc-system-sim/ni_hw_models.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/utils/rfnoc-system-sim/ni_hw_models.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2016 Ettus Research
 #
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/utils/rfnoc-system-sim/rfnocsim.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/utils/rfnoc-system-sim/rfnocsim.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2016 Ettus Research
 #
--- uhd-3.13.1.0.orig/fpga-src/usrp3/tools/utils/rfnoc-system-sim/sim_colosseum.py
+++ uhd-3.13.1.0/fpga-src/usrp3/tools/utils/rfnoc-system-sim/sim_colosseum.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2016 Ettus Research
 #
--- uhd-3.13.1.0.orig/fpga-src/usrp3/top/python/check_inout.py
+++ uhd-3.13.1.0/fpga-src/usrp3/top/python/check_inout.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/fpga-src/usrp3/top/python/check_timing.py
+++ uhd-3.13.1.0/fpga-src/usrp3/top/python/check_timing.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2011-2012 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/host/examples/python/benchmark_rate.py
+++ uhd-3.13.1.0/host/examples/python/benchmark_rate.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2018 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/examples/python/curses_fft.py
+++ uhd-3.13.1.0/host/examples/python/curses_fft.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2017-2018 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/examples/python/rx_to_file.py
+++ uhd-3.13.1.0/host/examples/python/rx_to_file.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2017-2018 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/examples/python/tx_waveforms.py
+++ uhd-3.13.1.0/host/examples/python/tx_waveforms.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2017-2018 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/lib/convert/gen_convert_general.py
+++ uhd-3.13.1.0/host/lib/convert/gen_convert_general.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2011-2012 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/deps/import_rpclib.py
+++ uhd-3.13.1.0/host/lib/deps/import_rpclib.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2017 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_ad5623_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_ad5623_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_ad7922_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_ad7922_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_ad9510_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_ad9510_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010-2012 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_ad9522_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_ad9522_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010-2011,2015 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_ad9777_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_ad9777_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_ad9862_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_ad9862_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_adf4350_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_adf4350_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_adf4351_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_adf4351_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_adf4360_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_adf4360_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_adf5355_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_adf5355_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_adf5356_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_adf5356_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2017 Ettus Research, A National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_ads62p44_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_ads62p44_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_ads62p48_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_ads62p48_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2013 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_lmx2592_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_lmx2592_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2017 Ettus Research, A National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_max2112_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_max2112_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_max2118_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_max2118_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_max2829_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_max2829_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_max2870_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_max2870_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2013 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_max2871_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_max2871_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2014 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_mgcpld_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_mgcpld_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2017 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_tda18272hnm_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_tda18272hnm_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/ic_reg_maps/gen_tuner_4937di5_regs.py
+++ uhd-3.13.1.0/host/lib/ic_reg_maps/gen_tuner_4937di5_regs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/rfnoc/nocscript/gen_basic_funcs.py
+++ uhd-3.13.1.0/host/lib/rfnoc/nocscript/gen_basic_funcs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 Generate the function list for the basic NocScript functions
 """
--- uhd-3.13.1.0.orig/host/lib/transport/gen_vrt_if_packet.py
+++ uhd-3.13.1.0/host/lib/transport/gen_vrt_if_packet.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010-2013,2015 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/transport/nirio/lvbitx/process-lvbitx.py
+++ uhd-3.13.1.0/host/lib/transport/nirio/lvbitx/process-lvbitx.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # Copyright 2013-2015 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/lib/usrp/dboard/twinrx/table_to_cpp.py
+++ uhd-3.13.1.0/host/lib/usrp/dboard/twinrx/table_to_cpp.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 import sys
 
--- uhd-3.13.1.0.orig/host/python/setup.py.in
+++ uhd-3.13.1.0/host/python/setup.py.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 #  Copyright 2017-2018 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/tests/devtest/benchmark_rate_test.py
+++ uhd-3.13.1.0/host/tests/devtest/benchmark_rate_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015-2016 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/tests/devtest/multi_usrp_test.py
+++ uhd-3.13.1.0/host/tests/devtest/multi_usrp_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2018 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/tests/devtest/python_api_test.py
+++ uhd-3.13.1.0/host/tests/devtest/python_api_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2018 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/host/tests/devtest/run_testsuite.py
+++ uhd-3.13.1.0/host/tests/devtest/run_testsuite.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015-2016 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/tests/devtest/rx_samples_to_file_test.py
+++ uhd-3.13.1.0/host/tests/devtest/rx_samples_to_file_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/tests/devtest/test_messages_test.py
+++ uhd-3.13.1.0/host/tests/devtest/test_messages_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/tests/devtest/test_pps_test.py
+++ uhd-3.13.1.0/host/tests/devtest/test_pps_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/tests/devtest/tx_bursts_test.py
+++ uhd-3.13.1.0/host/tests/devtest/tx_bursts_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/tests/devtest/uhd_test_base.py
+++ uhd-3.13.1.0/host/tests/devtest/uhd_test_base.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015-2016 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/tests/devtest/usrp_probe.py
+++ uhd-3.13.1.0/host/tests/devtest/usrp_probe.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015-2016 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/tests/devtest/usrp_probe_test.py
+++ uhd-3.13.1.0/host/tests/devtest/usrp_probe_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/utils/b2xx_side_channel.py
+++ uhd-3.13.1.0/host/utils/b2xx_side_channel.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2013-2015 Ettus Research LLC
--- uhd-3.13.1.0.orig/host/utils/converter_benchmark.py
+++ uhd-3.13.1.0/host/utils/converter_benchmark.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/utils/latency/graph.py
+++ uhd-3.13.1.0/host/utils/latency/graph.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2012 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/utils/latency/pci_hwdata.py
+++ uhd-3.13.1.0/host/utils/latency/pci_hwdata.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2013 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/utils/latency/run_tests.py
+++ uhd-3.13.1.0/host/utils/latency/run_tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2012 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/utils/usrp2_card_burner.py
+++ uhd-3.13.1.0/host/utils/usrp2_card_burner.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010-2011 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/utils/usrp2_card_burner_gui.py
+++ uhd-3.13.1.0/host/utils/usrp2_card_burner_gui.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010-2011 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/host/utils/usrp2_recovery.py
+++ uhd-3.13.1.0/host/utils/usrp2_recovery.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2010-2011 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/images/create_imgs_package.py
+++ uhd-3.13.1.0/images/create_imgs_package.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2014 Ettus Research LLC
 #
--- uhd-3.13.1.0.orig/images/populate_images.py
+++ uhd-3.13.1.0/images/populate_images.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2014 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/images/uhdimgs.py
+++ uhd-3.13.1.0/images/uhdimgs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2014 Ettus Research LLC
 # Copyright 2018 Ettus Research, a National Instruments Company
--- uhd-3.13.1.0.orig/mpm/python/setup.py.in
+++ uhd-3.13.1.0/mpm/python/setup.py.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2017 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/mpm/python/socket_test.py
+++ uhd-3.13.1.0/mpm/python/socket_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2017 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/tools/debs/convert_changelog.py
+++ uhd-3.13.1.0/tools/debs/convert_changelog.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2015 National Instruments Corp.
--- uhd-3.13.1.0.orig/tools/dissectors/make-dissector-reg.py
+++ uhd-3.13.1.0/tools/dissectors/make-dissector-reg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Looks for registration routines in the protocol dissectors,
 # and assembles C code to call all the routines.
--- uhd-3.13.1.0.orig/tools/gr-usrptest/apps/uhd_phase_alignment.py
+++ uhd-3.13.1.0/tools/gr-usrptest/apps/uhd_phase_alignment.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2018 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/tools/gr-usrptest/apps/usrp_fpga_funcverif.py
+++ uhd-3.13.1.0/tools/gr-usrptest/apps/usrp_fpga_funcverif.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2018 Ettus Research, a National Instruments Company
 #
--- uhd-3.13.1.0.orig/tools/gr-usrptest/apps/usrp_phasealignment.py
+++ uhd-3.13.1.0/tools/gr-usrptest/apps/usrp_phasealignment.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from gnuradio.uhd.uhd_app import UHDApp
 from usrptest.flowgraphs import phasealignment_fg
--- uhd-3.13.1.0.orig/tools/gr-usrptest/apps/usrp_selftest.py
+++ uhd-3.13.1.0/tools/gr-usrptest/apps/usrp_selftest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import argparse
 from usrptest import parsers
--- uhd-3.13.1.0.orig/tools/gr-usrptest/docs/doxygen/doxyxml/generated/compound.py
+++ uhd-3.13.1.0/tools/gr-usrptest/docs/doxygen/doxyxml/generated/compound.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 Generated Mon Feb  9 19:08:05 2009 by generateDS.py.
--- uhd-3.13.1.0.orig/tools/gr-usrptest/docs/doxygen/doxyxml/generated/compoundsuper.py
+++ uhd-3.13.1.0/tools/gr-usrptest/docs/doxygen/doxyxml/generated/compoundsuper.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 #
 # Generated Thu Jun 11 18:44:25 2009 by generateDS.py.
--- uhd-3.13.1.0.orig/tools/gr-usrptest/docs/doxygen/doxyxml/generated/index.py
+++ uhd-3.13.1.0/tools/gr-usrptest/docs/doxygen/doxyxml/generated/index.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 Generated Mon Feb  9 19:08:05 2009 by generateDS.py.
--- uhd-3.13.1.0.orig/tools/gr-usrptest/docs/doxygen/doxyxml/generated/indexsuper.py
+++ uhd-3.13.1.0/tools/gr-usrptest/docs/doxygen/doxyxml/generated/indexsuper.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 #
 # Generated Thu Jun 11 18:43:54 2009 by generateDS.py.
--- uhd-3.13.1.0.orig/tools/gr-usrptest/examples/lv_control_example.py
+++ uhd-3.13.1.0/tools/gr-usrptest/examples/lv_control_example.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from usrptest.labview_control import lv_control
 import time
--- uhd-3.13.1.0.orig/tools/gr-usrptest/python/flowgraphs/phasealignment_fg.py
+++ uhd-3.13.1.0/tools/gr-usrptest/python/flowgraphs/phasealignment_fg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 
 from gnuradio import gr
 from gnuradio import uhd
--- uhd-3.13.1.0.orig/tools/gr-usrptest/python/flowgraphs/selftest_fg.py
+++ uhd-3.13.1.0/tools/gr-usrptest/python/flowgraphs/selftest_fg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from gnuradio import blocks
 from gnuradio import gr
--- uhd-3.13.1.0.orig/tools/gr-usrptest/python/functions.py
+++ uhd-3.13.1.0/tools/gr-usrptest/python/functions.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 import numpy as np
 import time
 import copy
--- uhd-3.13.1.0.orig/tools/gr-usrptest/python/labview_control/lv_control.py
+++ uhd-3.13.1.0/tools/gr-usrptest/python/labview_control/lv_control.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 from labview_automation.client import LabVIEWClient
 
 class vst_siggen:
--- uhd-3.13.1.0.orig/tools/gr-usrptest/python/phase_calc_ccf.py
+++ uhd-3.13.1.0/tools/gr-usrptest/python/phase_calc_ccf.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2016 Ettus Research LLC.
--- uhd-3.13.1.0.orig/tools/gr-usrptest/python/qa_measurement_sink_f.py
+++ uhd-3.13.1.0/tools/gr-usrptest/python/qa_measurement_sink_f.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # 
 # Copyright 2016 Ettus Research LLC.
--- uhd-3.13.1.0.orig/tools/gr-usrptest/python/rts_tests/test_phasealignment.py
+++ uhd-3.13.1.0/tools/gr-usrptest/python/rts_tests/test_phasealignment.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2016 Ettus Research LLC.
--- uhd-3.13.1.0.orig/tools/gr-usrptest/python/setup.py
+++ uhd-3.13.1.0/tools/gr-usrptest/python/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 try:
     from setuptools import setup
 except ImportError:
--- uhd-3.13.1.0.orig/tools/mega_fft/mega_fft.py
+++ uhd-3.13.1.0/tools/mega_fft/mega_fft.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 ##################################################
 # GNU Radio Python Flow Graph
 # Copyright 2014 National Instruments
--- uhd-3.13.1.0.orig/tools/mega_fft/mega_fft_2ch.py
+++ uhd-3.13.1.0/tools/mega_fft/mega_fft_2ch.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 ##################################################
 # GNU Radio Python Flow Graph
 # Copyright 2014 National Instruments
--- uhd-3.13.1.0.orig/tools/uhd_txrx_debug_prints/uhd_txrx_debug_prints_graph.py
+++ uhd-3.13.1.0/tools/uhd_txrx_debug_prints/uhd_txrx_debug_prints_graph.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # Copyright 2013-2014 Ettus Research LLC
